src/compiler/factory/emitHelpers.ts

```diff
@@ -783,7 +783,9 @@ namespace ts {
         text: `
             var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
                 if (k2 === undefined) k2 = k;
-                Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
+                var oldDesc = Object.getOwnPropertyDescriptor(m, k);
+                var desc = ("get" in oldDesc) && !("value" in oldDesc) ? oldDesc : { enumerable: true, get: function() { return m[k]; } };

```

Comment:
Don't need parens, and don't need the `value` check (a descriptor produced by the runtime won't have both).

```suggestion
                var desc = "get" in oldDesc ? oldDesc : { enumerable: true, get: function() { return m[k]; } };
```

Reply:
My only concern would be an import from something exported like this:

```ts
// somefile.js
class Foo {
  counter = 0;
  get value() { return this.counter++; }
}
const inst = new Foo();
module.exports = inst;

// reexport.ts
export * from "somefile.js";

// main.ts
import { value } from "./reexport";
console.log(value, value);
```

The current live binding approach eventually forwards to `inst.value`, so the counter works correctly. Changing the live binding approach would break this (although its not spec compliant behavior to begin with).

Maybe `__createBinding` should check for the `__esModule` flag that we (and others) set on the module?

Reply:
In the case of:

> ```ts
> // somefile.js
> class Foo {
>   counter = 0;
>   get value() { return this.counter++; }
> }
> const inst = new Foo();
> module.exports = inst;
> ```

`oldDesc` will be `undefined` because the `value` getter is defined on the prototype of `inst`.

Because of this, `__createBinding(exports, require("./somefile.js"), "value")` will throw a `TypeError` at runtime because the `in` operator can’t be applied to `undefined`.

Reply:
In the case of:

> ```ts
> // somefile.js
> class Foo {
>   counter = 0;
>   get value() { return this.counter++; }
> }
> const inst = new Foo();
> module.exports = inst;
> ```

`oldDesc` will be `undefined` because the `value` getter is defined on the prototype of `inst`.

---

Because of this, `__createBinding(exports, require(\"./somefile.js\"), \"value\")` will throw a `TypeError` at runtime because the `in` operator can’t be applied to `undefined`.